src/spag/yaml_util.rs

```diff
@@ -109,3 +110,11 @@ pub fn get_nested_value<'a>(y: &'a Yaml, keys: &[&str]) -> Option<&'a Yaml> {
         None
     }
 }
+
+// If body can be serialized to JSON, return a pretty JSON string, otherwise return original string 
+pub fn pretty_json(resp_output: String) -> String {
```

Comment:
So if this function takes a string slice here (`resp_output: &str`) instead of a String, then you shouldn't have to make copies before passing strings in. You would just call `pretty_json(&string)` or `pretty_json(string.as_str())` instead.